test/suites/chaos/zonal_failure_test.go

```diff
@@ -0,0 +1,489 @@
+			targetInstances = append(targetInstances, *instance.InstanceId)
+		}
+	}
+
+	subnetARNs := getSubnetsInAZ(ctx, env, targetAZ)
+	By(fmt.Sprintf("Found %d subnets in AZ %s", len(subnetARNs), targetAZ))
+
+	template := &fis.CreateExperimentTemplateInput{
+		Actions: map[string]fistypes.CreateExperimentTemplateActionInput{
+			"stop-instances": {
+				ActionId: aws.String("aws:ec2:stop-instances"),
+				Parameters: map[string]string{
+					"startInstancesAfterDuration": "PT5M", // Start instances after 5 minutes
+				},
+				Targets: map[string]string{
+					"Instances": "target-instances",
+				},
+			},
+			"ec2-capacity-error": {
+				ActionId: aws.String("aws:ec2:api-insufficient-instance-capacity-error"),
+				Parameters: map[string]string{
+					"availabilityZoneIdentifiers": targetAZ,
+					"duration":                    "PT5M",
+					"percentage":                  failurePercentage,
+				},
+				Targets: map[string]string{
+					"Roles": "target-roles",
+				},
+			},
+			"disrupt-subnet": {
+				ActionId: aws.String("aws:network:disrupt-connectivity"),
+				Parameters: map[string]string{
+					"duration": "PT5M",
+					"scope":    "all",
+				},
+				Targets: map[string]string{
+					"Subnets": "target-subnets",
+				},
+			},
+		},
+		Targets: map[string]fistypes.CreateExperimentTemplateTargetInput{
+			"target-instances": {
+				ResourceType:  aws.String("aws:ec2:instance"),
+				SelectionMode: aws.String("ALL"),
+				ResourceArns: lo.Map(targetInstances, func(id string, _ int) string {
+					return fmt.Sprintf("arn:aws:ec2:%s:%s:instance/%s", env.Region, env.ExpectAccountID(), id)
+				}),
+			},
+			"target-roles": {
+				ResourceType:  aws.String("aws:iam:role"),
+				SelectionMode: aws.String("ALL"),
+				ResourceArns: []string{
+					fmt.Sprintf("arn:aws:iam::%s:role/KarpenterNodeRole-%s", env.ExpectAccountID(), env.ClusterName),
+				},
+			},
+			"target-subnets": {
+				ResourceType:  aws.String("aws:ec2:subnet"),
+				SelectionMode: aws.String("ALL"),
+				ResourceArns:  subnetARNs,
+			},
+		},
+		StopConditions: []fistypes.CreateExperimentTemplateStopConditionInput{
+			{
+				Source: aws.String("none"),
+			},
+		},
+		RoleArn:     aws.String(fisRoleArn),
+		Description: aws.String(fmt.Sprintf("Simulate AZ failure in %s", targetAZ)),
+	}
+
+	// Create experiment template
+	experimentTemplate, err := env.FISAPI.CreateExperimentTemplate(ctx, template)
+	Expect(err).NotTo(HaveOccurred())
+
+	return *experimentTemplate.ExperimentTemplate.Id
+}
+
+func startExperiment(ctx context.Context, env *awsenv.Environment, templateID string) *fistypes.Experiment {
+	experiment, err := env.FISAPI.StartExperiment(ctx, &fis.StartExperimentInput{
+		ExperimentTemplateId: aws.String(templateID),
+	})
+	Expect(err).NotTo(HaveOccurred())
+	return experiment.Experiment
+}
+
+func setupRoles(env *awsenv.Environment) {
+	// First, check if the FIS service-linked role exists
+	serviceRoleName := "AWSServiceRoleForFIS"
```

Comment:
Can we just set up all the necessary role in the GH actions as part of cluster creation?